components/ConnectWalletButton.tsx

```diff
@@ -38,14 +38,13 @@ const ConnectWalletButton = (props) => {
     providerUrl,
   ])
 
-  const [useDevnet, setUseDevnet] = useState(false)
-  const router = useRouter()
+  const [useDevnet, setUseDevnet] = useLocalStorageState('false')
   const handleToggleDevnet = () => {
     setUseDevnet(!useDevnet)
     if (useDevnet) {
-      router.push(`${window.location.pathname}`)
+      window.location.href = `${window.location.pathname}`
     } else {
-      router.push(`${window.location.href}?cluster=devnet`)
+      window.location.href = `${window.location.href}?cluster=devnet`
```

Comment:
This is problematic, because it will result in an invalid URL if there's already one querystring parameter in the current URL.

Reply:
Agreed. Will push an update to fix this.